src/fondant/pipeline/compiler.py

```diff
@@ -515,3 +517,167 @@ def _set_configuration(self, task, fondant_component_operation):
+        self._resolve_imports()
+
+    def _resolve_imports(self):
+        try:
+            import sagemaker
+            import sagemaker.processing
+            import sagemaker.workflow.pipeline
+            import sagemaker.workflow.steps
+
+            self.sagemaker = sagemaker
+
+        except ImportError:
+            msg = """You need to install the sagemaker extras to use the sagemaker compiler,\n
+                     you can install it with `pip install fondant[sagemaker]`"""
+            raise ImportError(
+                msg,
+            )
+
+    def build_command(
+        self,
+        metadata: Metadata,
+        arguments: t.Dict[str, t.Any],
+        dependencies: t.List[str] = [],
+    ) -> t.List[str]:
+        # add metadata argument to command
+        command = ["--metadata", f"'{metadata.to_json()}'"]
+
+        # add in and out manifest paths to command
+        command.extend(
+            [
+                "--output_manifest_path",
+                f"{metadata.base_path}/{metadata.pipeline_name}/{metadata.run_id}/"
+                f"{metadata.component_id}/manifest.json",
+            ],
+        )
+
+        # add arguments if any to command
+        for key, value in arguments.items():
+            if isinstance(value, (dict, list)):
+                command.extend([f"--{key}", f"'{json.dumps(value)}'"])
+            else:
+                command.extend([f"--{key}", f"'{value}'"])
+
+        # resolve dependencies
+        if dependencies:
+            for dependency in dependencies:
+                # there is only an input manifest if the component has dependencies
+                command.extend(
+                    [
+                        "--input_manifest_path",
+                        f"{metadata.base_path}/{metadata.pipeline_name}/{metadata.run_id}/"
+                        f"{dependency}/manifest.json",
+                    ],
+                )
+
+        return command
+
+    def compile(
+        self,
+        pipeline: Pipeline,
+        output_path: str,
+        instance_type: str = "ml.t3.medium",
+        role_arn: t.Optional[str] = None,
+    ) -> None:
+        """Compile a fondant pipeline to sagemaker pipeline spec and save it
+        to a specified output path.
+
+        Args:
+            pipeline: the pipeline to compile
+            output_path: the path where to save the Kubeflow pipeline spec.
+            instance_type: the instance type to use for the processing steps
+            (see: https://aws.amazon.com/ec2/instance-types/ for options).
+            role_arn: the role arn to use for the processing steps,
+            if none provided the `sagemaker.get_execution_role()` role will be used.
+        """
+        run_id = pipeline.get_run_id()
+        path = pipeline.base_path
+        pipeline.validate(run_id=run_id)
+
+        component_cache_key = None
+
+        steps: t.List[t.Any] = []
+
+        with tempfile.TemporaryDirectory(dir=os.getcwd()) as tmpdirname:
```

Comment:
the scripts are uploaded to s3 by sagemaker once we compile the pipeline that is why we save the files to a temp folder that only gets deleted at the end.